rcode-reader/src/main/java/eu/rssw/pct/elements/MethodParameter.java

```diff
@@ -14,31 +14,45 @@
   private static final int PARAMETER_HANDLE = 2;
   private static final int PARAMETER_BIND = 4;
 
-  private int extent;
+  private final int paramNum;
+  private final int extent;
+  private final int flags;
+  private final int parameterType;
+  private final int paramMode;
+  private final int dataType;
+  private final String dataTypeName;
   // private int crc;
-  private int flags;
-  private int parameterType;
-  private int paramMode;
-  private int dataType;
   // private int fullNameLength;
-  private int paramNum;
 
-  private String argumentName;
+  public MethodParameter(int num, String name, int type, int mode, int flags, int dataType, String dataTypeName,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS00107)